* [meta-security][PATCH] python3-fail2ban: fix compile issue on some hosts
@ 2022-04-01 16:54 Armin Kuster
  0 siblings, 0 replies; only message in thread
From: Armin Kuster @ 2022-04-01 16:54 UTC (permalink / raw)
  To: yocto

Use python3-native to use 2to3

Fix build issue on some hosts with this error:
 (result, consumed) = self._buffer_decode(data, self.errors, final)
 | UnicodeDecodeError: 'utf-8' codec can't decode byte 0xd8 in position 152: invalid continuation byte

Signed-off-by: Armin Kuster <akuster808@gmail.com>
---
 recipes-security/fail2ban/python3-fail2ban_0.11.2.bb | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/recipes-security/fail2ban/python3-fail2ban_0.11.2.bb b/recipes-security/fail2ban/python3-fail2ban_0.11.2.bb
index 4118732..96e17b7 100644
--- a/recipes-security/fail2ban/python3-fail2ban_0.11.2.bb
+++ b/recipes-security/fail2ban/python3-fail2ban_0.11.2.bb
@@ -9,6 +9,8 @@ HOMEPAGE = "http://www.fail2ban.org"
 LICENSE = "GPL-2.0-only"
 LIC_FILES_CHKSUM = "file://COPYING;md5=ecabc31e90311da843753ba772885d9f"
 
+DEPENDS = "python3-native"
+
 SRCREV ="4fe4ac8dde6ba14841da598ec37f8c6911fe0f64"
 SRC_URI = " git://github.com/fail2ban/fail2ban.git;branch=0.11;protocol=https \
         file://initd \
@@ -21,6 +23,11 @@ S = "${WORKDIR}/git"
 
 do_compile () {
     cd ${S}
+
+    #remove symlink to python3
+    # otherwise 2to3 is run against it
+    rm -f bin/fail2ban-python
+
     ./fail2ban-2to3
 }
 
-- 
2.25.1



^ permalink raw reply related	[flat|nested] only message in thread
only message in thread, other threads:[~2022-04-04 14:43 UTC | newest]

Thread overview: (only message) (download: mbox.gz / follow: Atom feed)
-- links below jump to the message on this page --
2022-04-01 16:54 [meta-security][PATCH] python3-fail2ban: fix compile issue on some hosts Armin Kuster

